server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java

```diff
@@ -665,9 +665,6 @@ public boolean delete(TemplateProfile profile) {
             // Remove template details
             templateDetailsDao.removeDetails(template.getId());
 
-            // Remove deploy-as-is details (if any)
```

Comment:
I think we atleast need to add a check if the template is used by existing VMs or not. If it is not used then I don't see a problem removing deployasis details.
And maybe preserve specific details if needed.

Reply:
added the check @shwstppr, please review it again.